### ISSUE TITLE
Function download_to_filename is not working when using service account file

### ISSUE THREAD

**Issue by amitripshtos** [googleapis/python-storage#261]
#### Environment details

  - OS type and version:
  - Python version: `Python 3.8.3`
  - pip version: `pip 20.1.1 from /usr/local/lib/python3.8/site-packages/pip (python 3.8)`
  - `google-cloud-storage` version: 
```text
Name: google-cloud-storage
Version: 1.31.0
Summary: Google Cloud Storage API client library
Home-page: https://github.com/googleapis/python-storage
Author: Google LLC
Author-email: googleapis-packages@google.com
License: Apache 2.0
Location: /usr/local/lib/python3.8/site-packages
Requires: google-resumable-media, google-auth, google-cloud-core
Required-by: kfp
```

#### Steps to reproduce

  1. Set environment variable GOOGLE_APPLICATION_CREDENTIALS=/path/to/your/sa/file.json
  2. Set environment variable CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=/path/to/your/sa/file.json
  3. Run the code I added and see the traceback.

#### Code example

```python
from google.cloud import storage
from urllib.parse import urlparse

file_path='gs://my-bucket/my_file.txt'
parsed_uri = urlparse(file_path, allow_fragments=False)
storage_client = storage.Client()
bucket = storage_client.bucket(parsed_uri.netloc)
blob = bucket.blob(parsed_uri.path[1:])
blob.download_to_filename('/tmp/test.txt')
```

#### Stack trace
```
Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None, status=None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/code/my_code.py", line 105, in my_func
    blob.download_to_filename(file_path)
  File "/usr/local/lib/python3.8/site-packages/google/cloud/storage/blob.py", line 1131, in download_to_filename
    self.download_to_file(
  File "/usr/local/lib/python3.8/site-packages/google/cloud/storage/blob.py", line 1037, in download_to_file
    self._do_download(
  File "/usr/local/lib/python3.8/site-packages/google/cloud/storage/blob.py", line 888, in _do_download
    download = klass(
TypeError: __init__() got an unexpected keyword argument 'checksum'
```

I can verify things are working smooth in 1.27.0
Thanks!

**Comment by HemangChothani** [comment]
@amitripshtos I have tried to reproduce with you given steps except step2, as it's not needed for google-cloud-storage, but not able to reproduce.
I think this error is related to `google-resumable-media`'s older version, please  make sure you have installed 'google-resumable-media==1.0.0'  version. If it doesn't help then please provide more info about the versions of libraries and the code.

**Comment by tseaver** [comment]
@amitripshtos Please show the output of `pip list` in the environment where you have installed `google-cloud-storage`.  As @HemangChothani indicates, the error you report indicates that the version of `google-resumable-media` is too old.

**Comment by amitripshtos** [comment]
Hey,
Looks like indeed my google-resumeable-media version is too old:
```
google-api-core          1.22.1
google-auth              1.21.0
google-cloud-bigquery    1.24.0
google-cloud-core        1.4.1
google-cloud-storage     1.27.0
google-resumable-media   0.5.1
googleapis-common-protos 1.52.0
```

However, it got installed automatically when using google-cloud-storage 1.30.0 - so maybe it's a good idea to set a limit on the package's requirements.

**Comment by HemangChothani** [comment]
@amitripshtos It is already added in package requirements as you can see here https://github.com/googleapis/python-storage/blob/cf0774aa8ffd45d340aff9a7d2236d8d65c8ae93/setup.py#L31-L34

But in your pip i can see google-cloud-storage==1.27.0 so with that google-resumable-media==0.5.1 is compaible.

Closing this issue for now, feel free to reopen if it occur again or answer is not satisfied.

**Comment by chanhlt12** [comment]
The error is still there.

Here is my `pip list` output:

```
google-api-core                    1.24.1
google-api-python-client           1.12.8
google-auth                        1.24.0
google-auth-httplib2               0.0.4
google-auth-oauthlib               0.4.2
google-cloud                       0.34.0
google-cloud-bigquery              2.6.1
google-cloud-core                  1.5.0
google-cloud-storage               1.35.0
google-crc32c                      1.1.0
google-resumable-media             1.2.0
googleapis-common-protos           1.52.0

```

**Comment by HemangChothani** [comment]
@chanhlt12 Could you please share the stack trace of error.

**Comment by chanhlt12** [comment]
@HemangChothani 
This is the stack trace.

```shell
Traceback (most recent call last):
  File "printing.py", line 290, in <module>
    pdf_file_path = download_original_pdf(uuid, file_name)
  File "printing.py", line 40, in download_original_pdf
    file.download_to_filename(file_path)
  File "/Users/admin/opt/anaconda3/lib/python3.8/site-packages/google/cloud/storage/blob.py", line 1183, in download_to_filename
    self.download_to_file(
  File "/Users/admin/opt/anaconda3/lib/python3.8/site-packages/google/cloud/storage/blob.py", line 1087, in download_to_file
    client.download_blob_to_file(
  File "/Users/admin/opt/anaconda3/lib/python3.8/site-packages/google/cloud/storage/client.py", line 712, in download_blob_to_file
    blob_or_uri._do_download(
  File "/Users/admin/opt/anaconda3/lib/python3.8/site-packages/google/cloud/storage/blob.py", line 948, in _do_download
    download = klass(
TypeError: __init__() got an unexpected keyword argument 'checksum'

```

However, the error only occurs with Anaconda3(python 3.8.5), there is no error when I switch to use Python 3.9.0(without Anaconda3).

**Comment by HemangChothani** [comment]
@chanhlt12 Please try to find the `google-resumable-media` version at the runtime. If you have installed package with conda then do `conda list`.
```python
import pkg_resources

pkg_resources.get_distribution("google-resumable-media").parsed_version
